component/metrics/scrape/scrape_test.go

```diff
@@ -16,7 +18,10 @@ import (
 func TestForwardingToAppendable(t *testing.T) {
 	l, err := logging.New(os.Stderr, logging.DefaultOptions)
 	require.NoError(t, err)
-	opts := component.Options{Logger: l}
+	opts := component.Options{
+		Logger:     l,
+		Registerer: &fakeRegisterer{},
```

Comment:
Once you're using the normal prometheus.Registerer interface, this can be `Registerer: prometheus.NewRegistry()` to avoid having to make a fake impl.